inc/ticket.class.php

```diff
@@ -6820,12 +6875,23 @@ function showTimeline($rand) {
             $date = $item_i['date_mod'];
          }
 
-         // check if curent item user is assignee or requester
-         $user_position = 'left';
-         if ((isset($ticket_users_keys[$item_i['users_id']])
-              && ($ticket_users_keys[$item_i['users_id']] == CommonItilActor::ASSIGN))
-             || ($item['type'] == 'Assign')) {
-            $user_position = 'right';
+         // set item position depending on field timeline_position
+         $user_position = 'left'; // default position
+         if (isset($item_i['timeline_position'])) {
+            switch ($item_i['timeline_position']) {
+               case self::TIMELINE_LEFT:
+                  $user_position = 'left';
+                  break;
+               case self::TIMELINE_MIDLEFT:
+                  $user_position = 'left middle';
+                  break;
+               case self::TIMELINE_MIDRIGHT:
+                  $user_position = 'right middle';
+                  break;
+               case self::TIMELINE_RIGHT:
+                  $user_position = 'right';
+                  break;
```

Comment:
maybe add a default case (TIMELINE_RIGHT), to prevent potential empty field, also, add an else statement (for the isset) to do the same thing

Reply:
default value is here: https://github.com/tomolimo/glpi/blob/ccda0aef3271788b34ee382c57af7de8bb20f3d7/inc/ticket.class.php#L6879

Reply:
ok right